pkg/operator/v1helpers/core_getters.go

```diff
@@ -100,3 +116,26 @@ func (g combinedSecretInterface) List(opts metav1.ListOptions) (*corev1.SecretLi
 	}
 	return ret, nil
 }
+
+func validateGetOptions(options metav1.GetOptions) error {
+	return validateResourceVersion(options.ResourceVersion)
+}
+
+func listOptionsToSelector(options metav1.ListOptions) (labels.Selector, error) {
+	if len(options.FieldSelector) > 0 || options.Watch || len(options.Continue) > 0 {
+		return nil, fmt.Errorf("list options against cache has unsupported fields: %v", options)
+	}
+	if err := validateResourceVersion(options.ResourceVersion); err != nil {
+		return nil, err
+	}
+	return labels.Parse(options.LabelSelector)
+}
+
+func validateResourceVersion(resourceVersion string) error {
+	switch resourceVersion {
+	case "", "0":
+		return nil
+	default:
+		return fmt.Errorf("options against cache with resource version are unsupported: %s", resourceVersion)
```

Comment:
nit: would the following be more readable? 

`fmt.Errorf("options against cache with resource version %q are unsupported", resourceVersion)`